File: zig/two-fer/two_fer.zig
const std = @import("std");

pub fn twoFer(buffer: []u8, name: ?[]const u8) anyerror![]u8 {
    const person = name orelse "you";
    const result = try std.fmt.bufPrint(buffer, "One for {s}, one for me.", .{person});
    return result;
}

File: fortran/two-fer/two_fer.f90
module two_fer
  implicit none

contains

  function twoFer(name) result(phrase)
    character(*), intent(in), optional :: name
    character(:), allocatable :: phrase
    character(len=10) :: local_name

    if (present(name)) then
        local_name = name
    else
        local_name = "you"
    end if
    phrase = "One for " // trim(local_name) // ", one for me."
  end function twoFer

end module two_fer
